impl/src/test/java/org/ehcache/internal/store/DefaultStoreProviderTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.internal.store;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.ehcache.config.ResourcePool;
+import org.ehcache.config.ResourcePools;
+import org.ehcache.config.ResourceType;
+import org.ehcache.config.ResourceUnit;
+import org.ehcache.config.units.EntryUnit;
+import org.ehcache.spi.cache.Store;
+import org.ehcache.spi.service.ServiceConfiguration;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class DefaultStoreProviderTest {
+  
+  @Rule
+  public ExpectedException expectedEx = ExpectedException.none();
+  
+  
+  @SuppressWarnings("unchecked")
+  @Test
+  public void testCreateStoreDiskPoolLessThanHeapPool() {
+    expectedEx.expect(IllegalArgumentException.class);
+    expectedEx.expectMessage("Disk pool size must be greater than Heap pool size");
```

Comment:
Personally I'm not a fan of this kind of exception assertion in junit.  The scoping of the `@Rule` field is wrong - it's applying to all methods when thats not really what you want.  From my point of view a simple try catch is simpler and just as clear, and can also be extremely specific about which line is expected to throw.